cmd/main_test.go

```diff
@@ -0,0 +1,268 @@
+}
+
+func performHTTPGetCall(url string, header HTTPheader) (*HTTPResponse, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		log.Printf("Error creating request: %s", err)
+		return nil, err
+	}
+	if header.name != "" {
+		req.Header.Add(header.name, header.value)
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Printf("Error making request: %s", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("Error reading response body:  %s", err)
+		return nil, err
+	}
+	response := &HTTPResponse{
+		Body:       string(body),
+		StatusCode: resp.StatusCode,
+	}
+	return response, nil
+}
+
+func TestCmd(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Main Suite")
+}
+
+var _ = Describe("Signup endpoint", func() {
+	Context("Calling the signup endpoint with GET", func() {
+		It("responds with ready and signedup", func() {
+			url := "http://localhost:5000/api/v1/signup"
+			expectedCode := http.StatusOK
+			expectedBody := `{"status":{"ready":true,"reason":"SignedUp"}}`
+
+			resp, err := performHTTPGetCall(url, HTTPheader{})
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unexpected error testing the \"%s\" endpoint: %v", url, err))
+			Expect(resp.StatusCode).To(Equal(expectedCode))
+			Expect(strings.TrimSpace(expectedBody)).To(Equal(strings.TrimSpace(resp.Body)))
+		})
+	})
+})
+
+var _ = DescribeTable("Workspace endpoint", func(header HTTPheader, expectedCode int, expectedBody string) {
+	url := "http://localhost:5000/workspaces"
+	resp, err := performHTTPGetCall(url, header)
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Unexpected error testing the \"%s\" endpoint: %v", url, err))
+	Expect(resp.StatusCode).To(Equal(expectedCode))
+	Expect(strings.TrimSpace(expectedBody)).To(Equal(strings.TrimSpace(resp.Body)))
+},
+	Entry(
+		"Calling the workspace endpoint for user1 responds only with the 'test-tenant' workspace info",
+		HTTPheader{"X-Email", "user1@konflux.dev"},
+		http.StatusOK,
+		`{"kind":"WorkspaceList","apiVersion":"toolchain.dev.openshift.com/v1alpha1","metadata":{},`+
+			`"items":[{"kind":"Workspace","apiVersion":"toolchain.dev.openshift.com/v1alpha1",`+
+			`"metadata":{"name":"test-tenant","creationTimestamp":null},"status":`+
+			`{"namespaces":[{"name":"test-tenant","type":"default"}]}}]}`),
+	Entry(
+		"Workspace endpoint for user2 responds with 2 namespaces info",
+		HTTPheader{"X-Email", "user2@konflux.dev"},
+		http.StatusOK,
+		`{"kind":"WorkspaceList","apiVersion":"toolchain.dev.openshift.com/v1alpha1","metadata":{},`+
+			`"items":[{"kind":"Workspace","apiVersion":"toolchain.dev.openshift.com/v1alpha1",`+
+			`"metadata":{"name":"test-tenant","creationTimestamp":null},"status":{"namespaces":`+
+			`[{"name":"test-tenant","type":"default"}]}},{"kind":"Workspace","apiVersion":"toolchain.dev.openshift.com/v1alpha1"`+
```

Comment:
Can we make this line nice as well?

Reply:
I find it not as nice when broken to more lines, but fixed.